src/poetry/puzzle/provider.py

```diff
@@ -570,6 +570,9 @@ def complete_package(
                         continue
                     self.search_for_direct_origin_dependency(dep)
 
+        active_extras = None if package.is_root() else dependency.extras
+        _dependencies = self._add_implicit_dependencies(_dependencies, active_extras)
```

Comment:
Since `_dependencies` is only used once, it's probably better to skip the variable assignment, by inlining in into the `_add_implicit_dependencies` call